modxcms/revolution
# Add resource alias in template picker as needed

### What does it do?
Conditionally includes the `alias` field when the `friendly_urls` is enabled and `automatic_alias` is disabled.

### Why is it needed?
Given the scenario above, a resource can not be saved via the template picker because the alias is not present.

### How to test

1. Clear all caches
2. Enable the template picker system setting (`enable_template_picker_in_tree`)
3. Toggle the `friendly_urls` and `automatic_alias` system settings in various combinations, each time performing a hard reload of the page
4. Create a new Resource via the add ("+") button in the Resources tree and verify the ability to save no matter what settings values you have chosen.

### Related issue(s)/PR(s)
Resolves #16178
